### ISSUE TITLE
Add support for IValueProvider

### ISSUE THREAD

**Issue by chiaramooney** [microsoft/react-native-windows#13328]
## Documentation
IValueProvider Documentation: https://learn.microsoft.com/en-us/windows/win32/api/uiautomationcore/nn-uiautomationcore-ivalueprovider

## Behavior
Components of control type Slider, ProgressBar, Hyperlink, and Button should have conditional support for the UIA IValueProvider (Documentation: https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-controlpatternmapping). Value for IValueProvider should be supplied by the `accessibilityValue` property (if this property was specified). See #12510 for issue for accessibilityValue specification. 

## Implementation Plan
We should:

Implement the IValueProvider APIs in CompositionDynamicAutomationProvider.
Set the list of controls who implement the IValueProvider to be Slider, ProgressBar, Hyperlink, and Button under certain conditions. To get more information on the conditions visit the page for the individual UIA control types.

For addition reference, see implementation of other provider on Fabric: https://github.com/microsoft/react-native-windows/pull/11874

## To Be Clarified 
Q: Are there scenarios where accessibilityValue should have a default value for a control, or should it only take a value when the `accessibilitValue` was specified?
A: It appears from the UIA documentation that there are no controls where an accessibility value is required. There are only controls where conditionally there should be an accessibility value.

## Implementation
| Control Type | Value Pattern Supported | IsReadOnly | Supports get_Value | Support set_Value |
| -- | -- | -- | -- | -- |
| TextInput | Yes | No | Yes | Yes |
| Searchbox | Yes | No | Yes | Yes |
| Adjustable | Yes | No | Yes | Not Yet, To-Do |
| Spinbutton | Yes | No | Yes | Not Yet, To-Do |
| Hyperlink | Yes | Yes | Yes | N/A |
| Button | Yes | Yes | Yes | N/A |
| ProgressBar | Yes | Yes | Yes | N/A |
| ComboBox | Yes | Yes | Yes | Not Yet, To-Do |
| ListItem | Yes | Yes | Yes | N/A |

**Comment by chiaramooney** [comment]
https://github.com/microsoft/react-native-windows/issues/12361

**Comment by chiaramooney** [comment]
https://github.com/microsoft/react-native-windows/issues/12510
